src/mol-theme/color/entity-id.ts

```diff
@@ -84,15 +97,15 @@ export function EntityIdColorTheme(ctx: ThemeDataContext, props: PD.Values<Entit
             let serial: number | undefined = undefined;
             if (StructureElement.Location.is(location)) {
                 const entityId = getEntityId(location);
-                const modelIndex = location.structure.models.indexOf(location.unit.model);
-                const k = key(entityId, modelIndex);
+                const sourceSerial = sourceSerialMap.get(location.unit.model.sourceData) ?? -1;
```

Comment:
Should this fallback to `location.structure.models.indexOf(location.unit.model)` instead of the `-1`?

Reply:
no, if it is not found there is some error and its better to show the DefaultColor.